firmware/main.ino

```diff
@@ -0,0 +1,67 @@
+// Program to control a robot on a cable
+
+#include <ArduinoJson.h>
+#include <StreamUtils.h>
+#include "interpreter.h"
+#include "json.h"
+#include "sensor.h"
+#include <SoftwareSerial.h>
+#include "object.h"
```

Comment:
Style tip: we don't have an official c++ style guide for PRS, but it's generally a good habit to adopt one informally for your own code. Most C++ style guides specify the order in which #include directives should be listed -- for example, here's one of Google's:

https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes